### PR TITLE
Replace cluster_context hash with a struct

### LINKED ISSUE #853: Pharos::Config#kube_client memoization breaks reconfiguration
`Pharos::Config.kube_client` memoization (assumedly) makes kubeclient reconfiguration crash.

### PR DESCRIPTION
**Pull request by kke**
Adds some robustness to the previously vague non-structured hash. Now there's a central place where all the cluster-context keys are listed.

Fixes #853 by refactoring the `kube_client` initialization

### CODE REVIEW

File: lib/pharos/phase.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@ def self.register_component(component)
     def initialize(host, config: nil, cluster_context: nil)
       @host = host
       @config = config
-      @cluster_context = cluster_context
+      @cluster_context = (cluster_context.is_a?(Hash) ? Pharos::Context.new(cluster_context) : cluster_context) if cluster_context
```
Comment (kke) [inline lib/pharos/phase.rb]: This is here to avoid changing specs

**Comment by jakolehm** [comment]
Conflicts again...

File: lib/pharos/context.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module Pharos
+  class Context < Pharos::Configuration::Struct
```
Comment (jakolehm) [inline lib/pharos/context.rb]: Perhaps a bit too generic name? How about `ClusterContext`?

File: lib/pharos/phases/load_cluster_configuration.rb [hunk 3]
```diff
@@ -26,7 +26,7 @@ def build_config(configmap)
 
       # @return [K8s::Resource, nil]
       def previous_config_map
-        kube_client.api('v1').resource('configmaps', namespace: 'kube-system').get('pharos-config')
+        kube_client.api('v1').resource('configmaps', namespace: 'kube-system').get('pharos-config').tap { |pvc| puts "-------------------"; puts pvc.inspect; puts "--------------" }
```
Comment (jakolehm) [inline lib/pharos/phases/load_cluster_configuration.rb]: What is this?

**Pull request closed without merging**